client/components/animated-icon/index.jsx

```diff
@@ -0,0 +1,33 @@
+import classNames from 'classnames';
+import lottie from 'lottie-web/build/player/lottie_light';
+import { useEffect, useRef } from 'react';
+
+const AnimatedIcon = ( { icon, className } ) => {
+	const iconEl = useRef();
+
+	useEffect( () => {
+		const reducedMotion = window.matchMedia( '(prefers-reduced-motion: reduce)' ).matches;
+
+		const animation = lottie.loadAnimation( {
+			container: iconEl.current,
+			renderer: 'svg',
+			loop: false,
+			autoplay: ! reducedMotion,
+			path: icon,
+		} );
+
+		if ( reducedMotion ) {
+			animation.addEventListener( 'config_ready', () => {
+				animation.goToAndPlay( animation.totalFrames, true );
+			} );
+		}
+
+		return () => animation.destroy();
+	}, [ icon ] );
+
+	return (
+		<div ref={ iconEl } className={ classNames( 'app-animated-icon__container', className ) }></div>
```

Comment:
Now that this its own component, I think it makes sense to have this be a top-level component class name. We could move the associated styles to `client/components/animated-icon/styles.scss`. WDYT? 
```suggestion
		<div ref={ iconEl } className={ classNames( 'animated-icon', className ) }></div>
```

Reply:
I've gone ahead to commit the suggestion, thanks! 

> We could move the associated styles to client/components/animated-icon/styles.scss. WDYT?

I don't believe there are currently any generic styles that all instances of this component would share. The existing instances have heights and widths defined in CSS, but these vary.

Do you think there's anything generic that could be added to `animated-icon/styles.scss`? Or should we possibly remove the static `animated-icon` class name altogether?

Reply:
Aha. I did not realize no styles are attached to the `animated-icon` selector. My initial thought was that we remove it if it is unused. However, it may provide some debugging value when viewing the source in a browser's developer tools. I'm fine with either retaining or removing the selector.